### PR TITLE
[depends] alsa-lib: fix typo in configure.in

### PR DESCRIPTION
**Pull request by XECDesign**
## Description

Change `${host-os}` to `${host_os}` in `tools/depends/target/alsa-lib`'s `configure.in`.

## Motivation and Context

Currently, if alsa-lib's configure finds `${host_cpu}-${host_os}-gcc`, it will set `$CC` to `${host_cpu}-${host-os}-gcc`. Since the variables are not the same (on the raspberry pi `${host-os}` is set to `arm-unknown`, resulting in `$CC` being a non-existent `arm-arm-unknown-..`), `configure` fails when testing the compiler.

## How Has This Been Tested?

On a raspberry pi:
```
cd tools/depends
./configure --with-platform=raspberry-pi --prefix=${HOME}/opt/kodi
make
```
Without the patch, configure claims the C compiler is not able to produce binaries. With the patch, the build completes.

## Types of change

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [X] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by MartijnKaijser** [comment]
@fritsch any idea why we use this ancient also lib?

**Comment by fritsch** [comment]
We don't use newer features :-) @wsnipex you'd prefer a bump instead?

**Comment by XECDesign** [comment]
Is it worthwhile opening a PR of changes required to make builds work like it says on the tin natively on ARM (on a pi3 or a chromebook running Raspbian, in my case)?

There is another change needed to get libssh to build, but it's a little less trivial. The issue is that cmake fails with:
> Could NOT find ZLIB (missing: ZLIB_LIBRARY ZLIB_INCLUDE_DIR)

I worked around the problem by modifying the Makefile to pass the necessary `-D` flags to cmake, but that's not portable. I haven't looked at exactly why cmake doesn't find and set those itself. It would be nice to have a place to put such changes and get a bit of review or suggestions from kodi devs.

Or should I just open a trac issue for that?

**Comment by wsnipex** [comment]
Can you please explain how you build?
Usually building natively (not cross compiling) doesn't use depends at all. And cross compiling with depends shouldn't hit that bug at all, since we explicitly set compilers and flags.

**Comment by XECDesign** [comment]
For now, I have only built depends.

The plan is to build kodi and set the appropriate `PATH` and `LD_PRELOAD_PATH` in the wrapper. I have been advised that Debian's packaging is frowned upon and not supported by kodi, that it's best to use the bundled dependencies where possible. and that the official kodi PPA is the way to go. Porting packages from the PPA back to something that works on Debian is proving to be challenging and still heavily relies on system libraries without kodi's patches.

I have run into issues with replacing the distro's own packages in order to meet the required dependencies. For example, libbluray jumps from 0.6.2 to 0.9.2. Many raspbian/debian packages are linked against it and I don't know whether it's entirely backwards compatible. libgif goes from 4.1.6 to 5.0.5. dcadec comes from an entirely different source.

That means that with each release, I would need to go over all of the dependencies changes, figure out which are safe to upgrade and what to do with those that aren't. I have no way of knowing if all the libraries are backwards compatible or whether the SOVERSION of each library is set correctly. I worry that this will only become more and more time consuming over time as new dependencies and patches are added and versions are bumped.

The short version is that I would like a kodi package for raspbian that has all the patches, most compatibility, but doesn't replace system packages and is more maintainable in the long term.

**Comment by wsnipex** [comment]
ok, makes sense, specially for debian stable based distros. Thanks for the explanation.
The PR here is ok and pending a successful build test from jenkins, will be merged.
For the libssh issue: can you either post a log showing the issue on the forum, or hop on IRC (freenode, #kodi-dev) and ping me?

jenkins build this please
